### PR TITLE
Add in ignore-filters option

### PR DESCRIPTION
**Pull request by MoT3rror**
Depends on https://github.com/composer/composer/pull/6062

Gives the user an option to ignore the HgExcludeFilter, GitExcludeFilter and ComposerExcludeFilter filters by using `ignore-filters` on the archive object.

Example:
```
{
  "homepage": "http:\/\/example.packages.com",
  "name": "Packages",
  "archive": {
    "directory": "storage\/dist",
    "format": "zip",
    "ignore-filters": true
  },
  "repositories": [
    {
      "type": "package",
      "package": {
        "name": "package\/package",
        "version": "1.0",
        "dist": {
          "type": "zip",
          "url": "https://github.com/package/package/archive/1.0.zip"
        }
      }
    }
  ]
}
```

### CODE REVIEW

**Comment by alcohol** [comment]
composer/composer#6062 is available as of `composer/composer:1.4.0`.

Can you update the `composer/composer` requirement in `composer.json` to match that?

File: src/Builder/ArchiveBuilder.php [hunk 1]
```diff
@@ -42,6 +42,7 @@ public function dump(array $packages)
         $format = isset($this->config['archive']['format']) ? $this->config['archive']['format'] : 'zip';
         $endpoint = isset($this->config['archive']['prefix-url']) ? $this->config['archive']['prefix-url'] : $this->config['homepage'];
         $includeArchiveChecksum = isset($this->config['archive']['checksum']) ? (bool) $this->config['archive']['checksum'] : true;
+        $ingoreFilters = isset($this->config['archive']['ignore-filters']) ? (bool) $this->config['archive']['ignore-filters'] : false;
```
Comment (glensc) [inline src/Builder/ArchiveBuilder.php]: spelling fix: `$ignoreFilters`

**Comment by glensc** [comment]
you should probably amend, rebase and force push your commits. unless maintainers use github squash merge feature. and creating PR from `master` branch is discouraged as well.

0.2¢

**Comment by alcohol** [comment]
Thanks

**Pull request merged**